packages/core/src/compiler.ts

```diff
@@ -171,7 +171,12 @@ export default class Compiler {
         this.setRuntimeFn('plural');
         break;
 
-      case 'function':
+      case 'function': {
+        const formatter = this.options.customFormatters?.[token.key];
```

Comment:
The option is always defined here.
```suggestion
        const formatter = this.options.customFormatters.[token.key];
```